### ISSUE TITLE
Generic Union formatting has changed

### ISSUE THREAD

**Issue by johnrom** [prettier/prettier#7655]
**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACA4guAnASzAwFUoDoBnDAQzzgyghg0ikoMvlgwHcCYACwxCGAAwBWlAB4AhPDTABrODADKNALZwAMgShwkGAGY0ANpThjaeAOYBXbbAB0GAKLStABzOGAOlAB6NgQEAAmAXDSXhB4LGxcGADCEJoxBrAA6gKCWKFhOAaEYJRGAEpwijDOAGJJADw1BHhchfhEAAp4EF6UAHwYALwYXt29QwPAARgYwapgAQC+ANwBQZh54ZHRsfFULClp0Agw2UKbBbjFpRgVVbUN0xhNLTBtxV09lAA0T2pwbEuRU6Y0oAQGw1GXwmGCmUBmcxgCygKzWUGCF22MTirH2yVS6ROZ1y+XeRBudzA1Tq9SeL1aVxB0IAZBgAIJMUR4elvRlgT69X7wjD-QFk-mg8FDEagmFwhGYeZLVaBdEbfJY3a49gHAnHLI5C7iimVKkPWnCnnigXUVkc5iCfBWvk2jCsgCaqjZtho+ntXOdwIlXyFM1F0CB7WDvSlkNlg0mT0RyNRquCshoEQMOxxCV1RwypxyGcj13Kpupj0tzQZQddzL+AIj1slUAhMuhCdhScVSOVARA3xAPRgFHYyFAdG6vA6dAQlGQIBoADcIAQwoOQAAjBTKVRqLyKfS2ZAwPD2OBDwQwTRmTKCARwSiHsBwf7sAQEZcCACei5KC5DvolhxF0NC2JoNDIKYFiXiAUhyLuKjqFouj6HA0HmJYQ4IWox6+AAivYzAYUgMHYSAh4tPgi4wD+XhPmAhBeDAm6jPoRZhEIyAABwAAxDlCliZAoXiLqMT74MuGFDgAjsR8A2ouNCUAAtAYcBhJpm70PJzRwGBEFQWRWFwZYmgEKe55mfhcBESRmGwUOMA0Fu2RcYIyAAEzOQoBBmMehyQYuTAGJu9iWAAKq5C4mU5IDLheACSUBabAahMQQLEcmEah0b4jmWIsixAA)
```sh
--parser typescript
```

**Input:**
```tsx

// Good
export const ComponentWithGoodGenerics: React.FC<FirstGenericProps> = props => {
  // etc
};

// Good
export const ComponentWithGoodGenerics: React.FC<
  FirstGenericProps,
  SecondGenericProps
> = props => {
  // etc
};

// Good
export const ComponentWithGoodGenerics: React.FC<
  FirstGenericProps & AnotherFirstGenericProps,
  SecondGenericProps
> = props => {
  // etc
};

// Good
export const ComponentWithGoodGenerics: React.FC<
  FirstGenericProps & AnotherFirstGenericProps & YetAgainAnotherFirstGenericProps,
  SecondGenericProps
> = props => {
  // etc
};

// Bad
export const ComponentWithBadGenerics: React.FC<
  FirstGenericProps &
  SecondGenericProps
> = props => {
  // etc
};

```

**Output:**
```tsx
// Good
export const ComponentWithGoodGenerics: React.FC<FirstGenericProps> = props => {
  // etc
};

// Good
export const ComponentWithGoodGenerics: React.FC<
  FirstGenericProps,
  SecondGenericProps
> = props => {
  // etc
};

// Good
export const ComponentWithGoodGenerics: React.FC<
  FirstGenericProps & AnotherFirstGenericProps,
  SecondGenericProps
> = props => {
  // etc
};

// Good
export const ComponentWithGoodGenerics: React.FC<
  FirstGenericProps &
    AnotherFirstGenericProps &
    YetAgainAnotherFirstGenericProps,
  SecondGenericProps
> = props => {
  // etc
};

// Bad
export const ComponentWithBadGenerics: React.FC<FirstGenericProps &
  SecondGenericProps> = props => {
  // etc
};

```

**Expected behavior:**

The closing brackets should always return to the originating indentation for consistency.

**Comment by sosukesuzuki** [comment]
Thank you for your report. Maybe you mean "intersection", not "union"(https://www.typescriptlang.org/docs/handbook/advanced-types.html). And this problem has been fixed via #6901 already, but not released yet. Please look [the playground result on #6901](https://deploy-preview-6901--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEB6VACA4hCATAHSjgA8AHCAJxg0igGcaBhCAWwuNgHUBLGACxz4sCOJR5h6SDACU4AQzAwAdADEmAHlU9KjEcXFgACpQhl6APgwBeDGVPmbV4EQwZ0GODDBEAvgG4iIg8hQmJyKho6RgwWdmgEGF4BUP0xCSlZBSU1TVcMbV0YNMMTM3oAGnyAZTg6PBKJMvMiK1t7cqcMFyg3Dy8fKACgqBDcMNIKalpoGLiOROTBccbJaTlFFXUNfMK9UVKHegwAMgwAQSgIATE94oOmo6rejFr61eb6Vps7I66evqYAZ+QJQYKYUJESaRGYMZhsBbcPjLYQPNZZTa5HYvO4fP5nS7Xfi3HT7AyPTpnACaXnOAHN5DwoISbpRcWjPs83G9oA0OUdvu0-tZnPl+t4QSMPAAheQTCLTaLw+KcJLI2V88nojY5ba7Un3LWfU41Oq8vHlQW-Toi7pioESob+EAVEBmGA8WbIUDySimADuRl9CHoyBA8gAbhAeHgXSAAEaURQAay81TIiiZdOQMEoAFc4K7+DBWAAbLj8PhwegZsBwWoMPg8CN8ACeYckoddTPoYhgJnkdNY8mQADN5KXe66AFb0EjSpNgVMwaryVhwAAyTLgY4nU5As5I1SzpbgAEU89cd0hx5PCyAM7oxGGYK2yNWwOIyDA4-YmWq8AEZAAA4AAZXQ6XsuCTMgw3sasxAjHdXQAR0veBPjDeR6AAWmIOA8AIuNKDgNCdDgAchxHG893vXtWB4HN8zok9z3Q69b33GB5HjXhAP4ZAACZXVzRlSyzOJhzDK5iDjPNewAFR40MaLvV0IwLABJKBCNgapPx4b9LjwapX1PXc718XwgA).

```ts
// Input
export const ComponentWithBadGenerics: React.FC<
  FirstGenericProps &
  SecondGenericProps
> = props => {
  // etc
};

// Output
export const ComponentWithBadGenerics: React.FC<
  FirstGenericProps & SecondGenericProps
> = props => {
  // etc
};
```

**Comment by johnrom** [comment]
Perfect, thanks!
